lib/goto.js

```diff
@@ -13,23 +13,21 @@ function error (name, message) {
  * @param {Goal} goal - The goal to execute.
  * @param {Callback} cb - The callback.
  */
-function goto (bot, goal, cb) {
-  function goalReached () {
-    cleanup()
-  }
-
-  function noPathListener (results) {
+const goto = (bot, goal, cb) => {
+  const noPathListener = results => {
     if (results.status !== 'noPath') return
     cleanup(error('NoPath', 'No path to the goal!'))
   }
 
-  function goalChangedListener (newGoal) {
+  const goalChangedListener = newGoal => {
     if (newGoal === goal) return
-    cleanup(error('GoalChanged', 'The goal was changed before it could be completed!'))
+    cleanup(
+      error('GoalChanged', 'The goal was changed before it could be completed!')
+    )
   }
 
-  function cleanup (err) {
-    bot.removeListener('goal_reached', goalReached)
+  const cleanup = err => {
+    bot.removeListener('goal_reached', cleanup)
```

Comment:
The goal reached event does pass a param to the callback, so using cleanup as the listener will wrongly throw an "error."

Reply:
oh, that is indeed correct, my bad